src/irmin-pack/s.ml

```diff
@@ -55,7 +55,34 @@ module type Specifics = sig
   (** [flush t] flush read-write pack on disk. Raises [RO_Not_Allowed] if called
       by a readonly instance.*)
 
-  val gc : ?unlink:bool -> repo -> commit_key -> unit
+  val start_gc :
+    ?unlink:bool ->
+    ?auto_finalise:bool ->
+    throttle:[ `Block | `Skip ] ->
+    repo ->
+    commit_key ->
+    bool Lwt.t
+  (** [start_gc] tries to start the gc process and returns true if the gc is
+      launched.
+
+      If [unlink] is false then temporary files on disk are kept after the gc
+      finished. This option is useful for debugging. The default is true.
+
+      If [auto_finalise] is true (the default) then there is no need to call
+      explicitly [finalise_gc].
```

Comment:
I suggest: 

If [auto_finalise] is true (the default) then it is not mandatory to call [finalise_gc]. The gc will automatically be finalised by the main process when [Repo.batch] is called (by the user or by irmin core). If the GC fails, [Repo.batch] with fail with the error.